### PR TITLE
fix(cli): large context causes E2BIG error during synthesis on Linux

### PR DESCRIPTION
**Pull request by otaviomacedo**
Instead of passing the context in an environment variable, the CLI now writes the context to a temporary file and sets an environment variable only with the location. The app then uses that location to read from the file.

Also tested manually on a Linux machine.

Fixes https://github.com/aws/aws-cdk/issues/19261.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/core/lib/app.ts [hunk 1]
```diff
@@ -142,16 +143,23 @@ export class App extends Stage {
...
-    for (const [k, v] of Object.entries(contextFromEnvironment)) {
+    const context = this.readContextFromTempFile() ?? this.readContextFromEnvironment() ?? {};
+    for (const [k, v] of Object.entries(context)) {
       this.node.setContext(k, v);
     }
   }
+
+  private readContextFromTempFile() {
+    const location = process.env[cxapi.CONTEXT_LOCATION_ENV];
+    return location != null ? fs.readJSONSync(location) : null;
```
Comment (rix0rrr) [inline packages/@aws-cdk/core/lib/app.ts:155]: We generally avoid using `null`, preferring `undefined` instead.

File: packages/@aws-cdk/core/lib/app.ts [hunk 2]
```diff
@@ -142,16 +143,23 @@ export class App extends Stage {
...
+  }
+
+  // for backward compatibility with old versions of the CLI
+  private readContextFromEnvironment() {
+    const contextJson = process.env[cxapi.CONTEXT_ENV];
+    return contextJson ? JSON.parse(contextJson) : null;
```
Comment (rix0rrr) [inline packages/@aws-cdk/core/lib/app.ts:161]: Same.

File: packages/aws-cdk/lib/api/cxapp/exec.ts [hunk 3]
```diff
@@ -44,7 +45,11 @@ export async function execProgram(aws: SdkProvider, config: Configuration): Prom
   context[cxapi.BUNDLING_STACKS] = bundlingStacks;
 
   debug('context:', context);
-  env[cxapi.CONTEXT_ENV] = JSON.stringify(context);
+
+  const contextDir = await fs.mkdtemp(path.join(os.tmpdir(), 'cdk-context'));
```
Comment (rix0rrr) [inline packages/aws-cdk/lib/api/cxapp/exec.ts:49]: Do we clean this directory up anywhere?

**Review by rix0rrr — APPROVED** [review]
Provisional approval

**Comment by RomainMuller** [comment]
This PR was reverted due to emergent regression test failures apparently related to it. Discussed with @otaviomacedo & provided support to re-create.

**Pull request merged**